### PR TITLE
Bulk upload for csv

### PR DESCRIPTION
**Pull request by brcre001**

### CODE REVIEW

File: breathecode/marketing/tasks.py [hunk 1]
```diff
@@ -362,3 +364,102 @@ def add_downloadable_slug_as_acp_tag(self, downloadable_id: int, academy_id: int
...
+        logger.error(message)
+
+    if not form_entry.location or not form_entry.academy:
+        message = 'No location or academy in form entry'
+        error_message += f'{message}, '
+        logger.error(message)
+
+    # Not needed because of Academy slug query problem
+
+    # if form_entry.academy and not re.findall(r'/^[a-z0-9]+(?:[a-z0-9-]+)+[a-z0-9]+$/', form_entry.academy):
```
Comment (jefer94) [inline breathecode/marketing/tasks.py:440]: remove the comments

File: breathecode/marketing/tests/tasks/tests_create_form_entry.py [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+        'won_at': None,
+        'contact_id': None,
+        'academy_id': None,
+        'user_id': None,
+        'lead_generation_app_id': None,
+        **data,
+    }
+
+
+def csv_upload_field(data={}):
```
Comment (jefer94) [inline breathecode/marketing/tests/tasks/tests_create_form_entry.py:135]: remove this function

File: breathecode/marketing/urls.py [hunk 3]
```diff
@@ -15,6 +15,8 @@
     path('app/lead', create_lead_from_app, name='app_lead'),
     path('lead/all', get_leads, name='lead_all'),  # TODO: HERE
     path('academy/lead', AcademyLeadView.as_view(), name='academy_lead'),
+    ### Testing UploadView ###
```
Comment (jefer94) [inline breathecode/marketing/urls.py:18]: remove this comment

**Review by jefer94 — CHANGES_REQUESTED** [review]

**Comment by jefer94** [comment]
Add /core.* to the `/.gitignore`

**Comment by jefer94** [comment]
Make sure the files CSV is generating during tests will be delete using the `tearDown` function inside the class of test

```py
class Test...(...):
    def tearDown(self):
        ...
    
    def test_.....
```

File: breathecode/monitoring/actions.py [hunk 4]
```diff
@@ -361,3 +361,54 @@ def download_csv(module, model_name, ids_to_download, academy_id=None):
         download.status_message = str(e)
         download.save()
         return False
+
+
+def upload_csv(module, model_name, file_content, academy_id=None):
```
Comment (jefer94) [inline breathecode/monitoring/actions.py:366]: Check if it is used, check if fix it or deleted if it is unused

**Comment by jefer94** [comment]
run `pipenv run format` to fix the format

File: breathecode/marketing/tasks.py (hunk 1 above)
Comment (jefer94) [inline breathecode/marketing/tasks.py:440]: Remove the comment

**Review by jefer94 — CHANGES_REQUESTED** [review]

File: breathecode/monitoring/actions.py (hunk 4 above)
Comment (brcre001) [reply breathecode/monitoring/actions.py:366]: In the issue https://github.com/4GeeksAcademy/About-4Geeks-Academy/issues/2576 dice que lo necesitamos?

File: breathecode/marketing/tests/urls/tests_upload.py [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+import pandas as pd
+from django.utils import timezone, dateparse
+
+UTC_NOW = timezone.now()
+
+
+class MarketingTestSuite(MarketingTestCase):
+    """Test /answer"""
+
+    def test_upload_without_auth(self):
```
Comment (jefer94) [inline breathecode/marketing/tests/urls/tests_upload.py:22]: add a tearDown to remove the csv using `os.remove(filename)`

**Comment by jefer94** [comment]
Make a pull request from development to fix the conflicts

File: breathecode/monitoring/actions.py (hunk 4 above)
Comment (jefer94) [reply breathecode/monitoring/actions.py:366]: I think under this conditions we can delete this function

**Review by jefer94 — DISMISSED** [review]

**Comment by jefer94** [comment]
Let me check the preview app

**Review by jefer94 — APPROVED** [review]

**Comment by jefer94** [comment]
https://breathecode-pr-589.herokuapp.com/

File: breathecode/marketing/views.py [hunk 6]
```diff
@@ -811,6 +819,90 @@ def delete(self, request, academy_id=None):
...
+
+        file_bytes = file_bytes.decode('utf-8')
+
+        with open(file.name, 'w') as f:
+            f.write(file_bytes)
+        print('Generic File: ', f)
+        df = pd.read_csv(file.name)
+        os.remove(file.name)
+        print('Data Frame: ', df)
+        required_fields = ['first_name', 'last_name', 'email', 'location', 'phone', 'language']
```
Comment (jefer94) [inline breathecode/marketing/views.py:860]: Create the admin (in `admin.py`) of `CSVUpload`

File: breathecode/marketing/views.py [hunk 7]
```diff
@@ -811,6 +819,90 @@ def delete(self, request, academy_id=None):
...
+
+        # Think about uploading correct files and leaving out incorrect ones
```
Comment (jefer94) [inline breathecode/marketing/views.py:862]: @alesanchezr the task is creating the `FormEntry` but if the data are duplicated, for example having the same row two times, is possible implement the same behavior of rest of `FormEntry` or validate the duplicated rows and prevent send it

**Review by jefer94 — CHANGES_REQUESTED** [review]

**Comment by jefer94** [comment]
The code work in preview app

**Comment by jefer94** [comment]
Fix tests

**Comment by jefer94** [comment]
Do not test the admin

**Comment by brcre001** [comment]
Create test for create_form_entry make sure to test for the data returned and see if *_id affects anything for the values that have the _id.

**Comment by jefer94** [comment]
Fix this tests
```bash
FAILED breathecode/marketing/tests/tasks/tests_create_form_entry.py::CreateFormEntryTestSuite::test_create_form_entry_with_dict_with_correct_format
FAILED breathecode/marketing/tests/urls/tests_upload.py::MarketingTestSuite::test_upload_random
```

Run `pipenv run format` to fix the format of code

**Pull request merged**